tests/test_controller_eui.py

```diff
@@ -0,0 +1,441 @@
+# This file is part of ts_m2com.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import sys
+import contextlib
+import unittest
+import pathlib
+
+from lsst.ts import tcpip
+from lsst.ts import salobj
+from lsst.ts.m2com import (
+    MockServer,
+    Controller,
+    CommandStatus,
+    CommandScript,
+    PowerType,
+    DigitalOutput,
+    collect_queue_messages,
+    get_queue_message_latest,
+    TEST_DIGITAL_OUTPUT_POWER_COMM,
+    TEST_DIGITAL_OUTPUT_POWER_COMM_MOTOR,
+)
+
+
+class TestControllerEui(unittest.IsolatedAsyncioTestCase):
+    """Test the Controller class for the engineering user interface (EUI).
+
+    This test case will be removed in the future after unifying the state
+    machines in the cell controller.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.config_dir = pathlib.Path(__file__).parents[0]
+        cls.host = tcpip.LOCAL_HOST
+        cls.timeout_in_second = 0.05
+
+        logging.basicConfig(
+            level=logging.INFO, handlers=[logging.StreamHandler(sys.stdout)]
+        )
+        cls.log = logging.getLogger()
+
+    @contextlib.asynccontextmanager
+    async def make_server(self):
+        """Instantiate the mock server of M2 for the test."""
+
+        server = MockServer(
+            self.host,
+            timeout_in_second=self.timeout_in_second,
+            port_command=0,
+            port_telemetry=0,
+            log=self.log,
+            is_csc=False,
+        )
+        server.model.configure(self.config_dir, "harrisLUT")
+        await server.start()
+
+        try:
+            yield server
```

Comment:
I think you may want to do:

```
try:
    yield server
    await server.close()
exception Except:
    # log exception here?
```

This avoid the situation where closing the server raises and exception and weird things happen.

Reply:
I only use this in the test scripts and `finally` is used to clean up all the resources (it will be executed with or without the `exception`). The `close()` in server code was designed to be safe to call even there is no connection. Therefore, I think it should be fine at this moment. Thanks!